src/dos/program_more.cpp

```diff
@@ -0,0 +1,508 @@
+		previous_column = current_column;
+
+		// Update new line counter, decide if pause needed
+		if (new_line && current_row)
+			++line_counter;
+		if (line_counter < max_lines)
+			continue;
+
+		// New line occured just enough times for a pause
+		decision = PromptUser();
+		if (decision == Decision::Terminate || decision == Decision::NextFile)
+			break;
+	}
+
+	if (GetCurrentColumn()) {
+		++line_counter;
+		WriteOut("\n");
+	}
+
+	return decision;
+}
+
+MORE::Decision MORE::PromptUser()
+{
+	line_counter = 0;
+	const bool multiple_files = input_files.size() > 1;
+
+	if (GetCurrentColumn())
+		WriteOut("\n");
+
+	if (multiple_files)
+		WriteOut(MSG_Get("SHELL_CMD_MORE_PROMPT_MULTI"));
+	else
+		WriteOut(MSG_Get("SHELL_CMD_MORE_PROMPT_SINGLE"));
+
+	auto decision = Decision::Terminate;
+	while (!shutdown_requested) {
+		CALLBACK_Idle();
+
+		uint16_t count = 1;
+		char choice    = 0;
+		DOS_ReadFile(STDIN, reinterpret_cast<uint8_t *>(&choice), &count);
+
+		if (count == 0 || choice == code_ctrl_c || choice == code_esc ||
+		    choice == 'q' || choice == 'Q')
+			break;
+
+		if (choice == code_cr || choice == ' ') {
+			decision = Decision::More;
+			break;
+		}
+
+		if (!multiple_files)
+			continue;
+
+		if (choice == 'n' || choice == 'N') {
+			decision = Decision::NextFile;
+			break;
+		}
+	}
+
+	if (decision == Decision::Terminate || decision == Decision::NextFile) {
+		WriteOut(" ");
+		WriteOut(MSG_Get("SHELL_CMD_MORE_TERMINATE"));
+		WriteOut("\n");
+		++line_counter;
+	} else {
+		// We are going to continue - erase the prompt
+		WriteOut("\033[M"); // clear line
+		auto counter = GetCurrentColumn();
+		while (counter--)
+			WriteOut("\033[D"); // cursor one position back
+	}
+
+	return decision;
+}
+
+bool MORE::GetCharacter(char &code)
+{
+	// clang-format off
+	constexpr std::array<char, 7> break_sequence = {
+		// CTRL + ZXCVBNM - emergency break key sequence, in case
+	    // user executes MORE.COM with STDIN as input; supported
+	    // at least by MS-DOS 6.22, DR-DOS 7.03 and PC-DOS 2000
+	    0x1a, 0x18, 0x03, 0x16, 0x02, 0x0e, 0x0d
+	};
+	// clang-format on
+
+	if (!tab_left) {
+		while (true) {
+			// Retrieve character from input stream
+			uint16_t count = 1;
+			DOS_ReadFile(input_handle,
+			             reinterpret_cast<uint8_t *>(&code),
+			             &count);
+			if (!count) {
+				break_idx = 0;
+				return false; // end of stream
+			}
+
+			// Handle emergency break sequence
+			if (break_enable && (code == break_sequence[break_idx])) {
+				if (++break_idx == break_sequence.size()) {
+					break_idx = 0;
+					return false; // emergency break
+				}
+			} else
+				break_idx = 0;
+
+			// Skip CR/LF characters if requested
+			if (skip_next_cr && code == code_cr)
+				skip_next_cr = false;
+			else if (skip_next_lf && code == code_lf)
+				skip_next_lf = false;
+			else
+				break;
+		}
+
+		// If TAB found, replace it with given number of spaces
+		if (code == '\t')
+			tab_left = tab_size;
+	}
```

Comment:
Suggest bracketing `{ }` all the single-line if/else sub-branches once you're inside a top-level bracketed branch.

(Once things get "brackety", use brackets all the time for nested content. This makes it easier to follow when closing multiple levels,  otherwise the closing bracket(s) can appear to be floating floating when there are several levels of non-bracketted nested levels -- like on line  444).

Reply:
OK, changed.